### PR TITLE
MuJoCo 2.1 compatible gym_compete

### PR DESCRIPTION
**Pull request by PavelCz**
- Update to muojco-py 2.1 which is the latest version and supports MuJoCo 2.1
- Remove copy of `mujoco_env` from `openai/gym` in favor of using the `openai/gym` version directly
- Remove `policy.py`. Not sure if this is desired, the main reason is that the current `policy.py `is only compatible with `tf`<2 and would need to be reworked. Older versions of `tf` might also limit the version of `gym` which could be used

### CODE REVIEW

**Comment by PavelCz** [comment]
Looks like I can't add you as a reviewer @AdamGleave.

**Comment by PavelCz** [comment]
Reminder to not merge this into `master`. AFAICT there is no way for me to set up the PR such that it gets merged into a new brach that doesn't exist yet, but I assume as admin you have that ability.

File: README.md [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
 
+This branch updates the HumanCompatibleAI fork to make gym_compete compatible with MuJoCo 2.1.
+
+List of changes:
+- Remove Python 2 and 3 compatibility code with six.
+- Updates for mujoco_py changes
+  - `env.model` -> `env.sim`
+  - Some attributes were renamed.
+- Update for compatiblity with newer version of `numpy`.
+- Some agents clipped actions before calculating rewards, some didn't. Now there is consistently no clipping.
```
Comment (AdamGleave) [inline README.md]:
```suggestion
- Some agents clipped actions before calculating rewards, some didn't. Now there is consistently no clipping in this codebase, although it is still clipped to actuator bounds inside `MujocoEnv`.
```

File: README.md [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
+- The `numarrows` attribute in xml was removed due to it having no function.
+- Remove copied code of `openai/gym`'s `mujoco_env` in favor of importing from gym.
```
Comment (AdamGleave) [inline README.md]:
```suggestion
- Remove copied code of `openai/gym`'s `mujoco_env` in favor of importing from `gym`.
```

File: README.md [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
+- Remove `policy_py` which relies on `tensorflow<2`
+- Remove the `demo_tasks.sh` script which wasn't runnable due to prior changes in the HumanCompatibleAI fork.
```
Comment (AdamGleave) [inline README.md:17]: Can you cherry-pick this commit, make a separate PR, and then we'll merge that to master (and then merge this PR to a separate branch)? We shouldn't have unrannable code in master either...

File: README.md [hunk 4]
```diff
@@ -19,15 +31,10 @@ Next install `gym-compete` package as:
...
 pip install -e .
 ```
-Check install is successful by coming out of the directory and trying `import gym_compete` in python console. Some users might require a `sudo pip install`.
 
-## Trying the environments
-Agent policies are provided for the various environments in folder `agent-zoo`. To see a demo of all the environments do:
+Or use pip to install / add to requirements as
 ```bash
-bash demo_tasks.sh all
+gym_compete @ git+https://github.com/HumanCompatibleAI/multiagent-competition.git@v0.1.0
```
Comment (AdamGleave) [inline README.md]: Will this install the right branch? I think we might need something like `@mujoco21` (if new branch is called `mujoco21`)

File: README.md [hunk 5]
```diff
@@ -19,15 +31,10 @@ Next install `gym-compete` package as:
...
 ```
-To instead try a single environment use:
-```bash
-bash demo_tasks.sh <task>
-```
-where `<task>` is one of: `run-to-goal-humans`, `run-to-goal-ants`, `you-shall-not-pass`, `sumo-ants`, `sumo-humans` and `kick-and-defend`
+Check install is successful by coming out of the directory and trying `import gym_compete` in python console. Some users might require a `sudo pip install`.
```
Comment (AdamGleave) [inline README.md]:
```suggestion
Check install is successful by coming out of the directory and trying `import gym_compete` in python console.
```
They should not if they're using virtualenvs correctly...

File: README.md [hunk 6]
```diff
@@ -1,16 +1,28 @@
...
+List of changes:
+- Remove Python 2 and 3 compatibility code with six.
+- Updates for mujoco_py changes
+  - `env.model` -> `env.sim`
+  - Some attributes were renamed.
+- Update for compatiblity with newer version of `numpy`.
+- Some agents clipped actions before calculating rewards, some didn't. Now there is consistently no clipping.
+- The `numarrows` attribute in xml was removed due to it having no function.
+- Remove copied code of `openai/gym`'s `mujoco_env` in favor of importing from gym.
+- Remove `policy_py` which relies on `tensorflow<2`
```
Comment (AdamGleave) [inline README.md]:
```suggestion
- Remove `policy.py` which relies on `tensorflow<2`
```

File: gym_compete/new_envs/agents/humanoid.py [hunk 7]
```diff
@@ -4,7 +4,8 @@
 
 
 def mass_center(mass, xpos):
-    return (np.sum(mass * xpos, 0) / np.sum(mass))[0]
+    # I think the reshape is necessary because np got stricter about broadcasting dims of size 1 vs dims of size 0, if not this calculation might be wrong
```
Comment (AdamGleave) [inline gym_compete/new_envs/agents/humanoid.py]: This comment doesn't inspire confidence :) Following line looks correct to me, if you're not sure can you test it a bit?

**Review by AdamGleave — CHANGES_REQUESTED** [review]
Thanks for the PR, some minor changes suggested.

I've updated the base branch (had to push new branch first to make it an option, but I don't think you have write access to this repo so you wouldn't have been able to do that).

File: README.md (hunk 4 above)
Comment (PavelCz) [reply README.md]: Yes, you're right that's what it should be called.

File: README.md (hunk 3 above)
Comment (PavelCz) [reply README.md:17]: Done

**Comment by PavelCz** [comment]
For posterity:
The changes in `mass_center()` are necessary because in some version prior to mujoco 2.1 `body_mass` was changed from 1d matrix (shape (13,1)) to vector (shape (13,)). Multiplying (`*`) this with a matrix (shape (13,3)) will fail on the vector. The reason is that numpy considers vectors as [1,n] arrays as opposed to [n,1] arrays during broadcasting.

File: gym_compete/new_envs/agents/humanoid.py (hunk 7 above)
Comment (PavelCz) [reply gym_compete/new_envs/agents/humanoid.py]: I investigated this, see comment below

**Comment by PavelCz** [comment]
@AdamGleave I addressed all comments.
Stephen also found a bug which I fixed in [this commit](https://github.com/HumanCompatibleAI/multiagent-competition/pull/6/commits/3620b978d6ee88699b46a3bcf2d9d1282fca8594) (now also part of this PR).

**Review by AdamGleave — APPROVED** [review]
LGTM

**Pull request merged**